public/app/plugins/panel/geomap/editor/MapCenterEditor.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, useMemo } from 'react';
+import { GrafanaTheme, StandardEditorProps } from '@grafana/data';
+import { Select, stylesFactory, useStyles } from '@grafana/ui';
+import { GeomapPanelOptions, MapCenterConfig } from '../types';
+import { centerPointRegistry, MapCenterID } from '../view';
+import { css } from '@emotion/css';
+import { NumberInput } from '../components/NumberInput';
+
+export const MapCenterEditor: FC<StandardEditorProps<MapCenterConfig, any, GeomapPanelOptions>> = ({
+  value,
+  onChange,
+  context,
+}) => {
+  const style = useStyles(getStyles);
+
+  const views = useMemo(() => {
+    const ids: string[] = [];
+    if (value?.id) {
+      ids.push(value.id);
+    } else {
+      ids.push(centerPointRegistry.list()[0].id);
+    }
+    return centerPointRegistry.selectOptions(ids);
+  }, [value?.id]);
+
+  return (
+    <div>
+      <Select
+        options={views.options}
+        value={views.current}
+        onChange={(v) => {
+          onChange({
+            id: v.value!,
+          });
+        }}
+      />
```

Comment:
Can this editor be created with just `addSelect` instead? And then the coordinates editor would become a conditional option(s)

Reply:
Maybe -- but I was hoping for something a little more fancy that knows what the current map view is and can explicitly get them in sync.